packages/sdk/client/src/services/agent-hosting-provider.ts

```diff
@@ -57,10 +126,114 @@ export class DXOSAgentHostingProviderClient implements AgentHostingProviderClien
     };
   }
 
+  public requestInitWithAuthToken(req: RequestInit): RequestInit {
+    return {
+      ...req,
+      headers: {
+        ...req.headers,
+        Authorization: 'Bearer ' + this._authToken,
+      },
+    };
+  }
+
+  public async checkEligibility(authToken: any) {
+    // TODO: Check auth token from CF worker whether identity is allowed to create agent.
+    return true;
+  }
+
+  @synchronized
+  async _ensureAuthenticated() {
+    if (this._authToken) {
+      return;
+    }
+    const { deviceKey } = this._halo.device!;
+
+    invariant(deviceKey, 'deviceKey not found');
+    const authDeviceCreds = await this._queryCredentials('dxos.halo.credentials.AuthorizedDevice', (cred) =>
+      PublicKey.equals(cred.subject.id, deviceKey),
+    );
+
+    invariant(authDeviceCreds.length === 1, 'Improper number of authorized devices');
+    invariant(this._rpc, 'RPC not initialized');
+    // TODO: factor out RPC operations and state management
+    if (this._rpcState === 'disconnected') {
+      await this._rpc.open();
+    }
+    await this._agentManagerAuth(authDeviceCreds[0]);
+  }
+
+  async _agentManagerAuth(authDeviceCreds: Credential) {
```

Comment:
Good to have a 1 sentence comment for this critical fn. (_authAgentManager?)

E.g., Initiates the auth sequence to retrieve the public key, then presents the credential...